noxfile.py

```diff
@@ -293,3 +307,27 @@ def format(session):
     session.run("isort", *LINT_FILES)
     for file in LINT_FILES:
         remove_trailing_whitespace(file)
+
+
+@nox.session
+def clean_build(session):
+    """Remove temporary files and building folders"""
+
+    # remove building folders and the nox cache
+    for i in ["docs/src/examples/", "docs/build", ".nox"]:
```

Comment:
why `.nox`? this will remove itself when running =/

Reply:
@ceriottm I think you added that piece

Reply:
Even without michele comment I kind of agree that removing folders like `.nox` is the point of this function

Reply:
hmm, I see this more as a way to clean temporary files created by the example, not to clean the virtualenvs. Also `rm -rf .nox` is trivial compared to what `nox -e clean` does, so leaving them separate is not too much of a burden on the end user.

Reply:
So, the rationale for this whole PR is to have a setup similar to the clean_python script of other projects, wiping all build and caches. Upon checking, it's true that all of those don't wipe .tox so even though I added it in the first place, I now agree that for consistency we should not (or we should do this consistently everywhere). Removing it for now (and merging)